File: TestApp/Views/CascadingState.cs
namespace TestApp.Views;

using Generated;

public class CascadingState : CascadingStateBase
{
    public CascadingState(IServiceProvider sp) : base(sp)
    {
        loggedIn = new()
        {
            Id = 1,
            FirstName = "John"
        };
    }

    override public void SetLoggedIn(Account loggedIn)
    {
        this.loggedIn = loggedIn;
    }
}

File: TestApp.Web/wwwroot/js/CascadingState.js
var CascadingState = {
    setLoggedIn: async (el, loggedIn) => {
        var cascade = el.closest("._CascadingState");
        await dispatch(cascade, "SetLoggedIn", {
            loggedIn: loggedIn
        });
    },
    logIn: async (el) => await CascadingState.setLoggedIn(el, { Id: 1, FirstName: "John" }),
    logOut: async (el) => await CascadingState.setLoggedIn(el, {})
};
